### PR TITLE
Illinois Child Tax Credit

### LINKED ISSUE #4596: Illinois 2024 Child Tax Credit
[Reference](https://www.niskanencenter.org/proposed-illinois-child-tax-credit-is-a-step-in-the-right-direction/)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4596

### CODE REVIEW

File: policyengine_us/parameters/gov/contrib/states/il/child_tax_credit/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: Illinois provides the following child tax credit amount for each individual child.
+metadata:
+  unit: currency-USD
+  period: year
+  label: Illinois Child Tax Credit amount
+  reference:
+    - title: Niskanen Center Proposed Illinois Child Tax Credit is a step in the right direction
+      href: https://www.niskanencenter.org/proposed-illinois-child-tax-credit-is-a-step-in-the-right-direction/
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/contrib/states/il/child_tax_credit/amount.yaml:8]: need bill ref

**Review by MaxGhenis — COMMENTED** [review]
gov expected to sign it today, let's move out of contrib https://abc7chicago.com/post/illinois-governor-jb-pritzker-sign-53-billion-budget/14913775/

**Comment by PavelMakarchuk** [comment]
> gov expected to sign it today, let's move out of contrib https://abc7chicago.com/post/illinois-governor-jb-pritzker-sign-53-billion-budget/14913775/

Will do

File: policyengine_us/variables/gov/states/il/tax/income/credits/il_child_tax_credit.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from policyengine_us.model_api import *
+
+
+class il_child_tax_credit(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/il/tax/income/credits/il_child_tax_credit.py:4]: More aligned to other states right?
```suggestion
class il_ctc(Variable):
```

File: policyengine_us/variables/gov/states/il/tax/income/credits/il_child_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    definition_period = YEAR
+    reference = "https://www.ilga.gov/legislation/fulltext.asp?DocName=&SessionId=112&GA=103&DocTypeId=HB&DocNum=4917&GAID=17&LegID=152789&SpecSess=&Session="
+    defined_for = StateCode.IL
+
+    def formula(tax_unit, period, parameters):
+        earned_income = tax_unit("tax_unit_earned_income", period)
+        agi = tax_unit("adjusted_gross_income", period)
+        larger_income = max_(earned_income, agi)
+        p = parameters(period).gov.states.il.tax.income.credits.ctc
+        amount = p.amount
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/il/tax/income/credits/il_child_tax_credit.py:18]: remove and use directly in formula

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**